src/gsi/radarZ_module.f90

```diff
@@ -0,0 +1,2889 @@
+!
+!-----------------------------------------------------------------------
+! Force explicit declarations.
+!-----------------------------------------------------------------------
+
+  IMPLICIT NONE
+
+!-----------------------------------------------------------------------
+! Variables can be changed by parameter retrieval
+!-----------------------------------------------------------------------
+  fos = 0.5              ! Maximum fraction of rain-snow mixture
+  foh = 0.3              ! Maximum fraction of rain-hail mixture
+  fog = 0.0              ! Maximum fraction of rain-hail mixture
+      
+  END SUBROUTINE init_fox_no_grpl
+
+
+  SUBROUTINE init_fox_no_hail() 
+
+!-----------------------------------------------------------------------
+!
+! PURPOSE:  
+!
+!  Setup default maximum fraction of water in the melting ice 
+!  when hail is suprressed. 
+!
+!-----------------------------------------------------------------------
+!
+! AUTHOR: Bryan Putnam, 12/14/10
+!
+!-----------------------------------------------------------------------
+! Force explicit declarations. 
+!-----------------------------------------------------------------------
+
+  IMPLICIT NONE 
+
+!-----------------------------------------------------------------------
+! Variables can be changed by parameter retrieval 
+!-----------------------------------------------------------------------
+
+  fos = 0.5             ! Maximum fraction of rain-snow mixture
+  foh = 0.0             ! Maximum fraction of rain-hail mixture
+  fog = 0.3             ! Maximum fraction of rain-hail mixture
+
+  END SUBROUTINE init_fox_no_hail
+
+  SUBROUTINE model_dsd(n0rain,n0snow,n0hail,n0grpl,rhosnow,rhohail,rhogrpl, &
+             alpharain,alphasnow,alphagrpl,alphahail)
+
+!-----------------------------------------------------------------------
+!
+! PURPOSE:
+!
+! Set dsd values to those used in the arps forecasts
+!
+!-----------------------------------------------------------------------
+!
+! AUTHOR:  Youngsun Jung, 1/20/2006
+!
+!-----------------------------------------------------------------------
+! Force explicit declarations.
+!-----------------------------------------------------------------------
+
+  IMPLICIT NONE
+
+!-----------------------------------------------------------------------
+! Define variables:
+!-----------------------------------------------------------------------
+
+
+  REAL(r_kind) :: n0rain,n0snow,n0hail,n0grpl,rhosnow,rhohail,rhogrpl,   &
+          alpharain,alphasnow,alphagrpl,alphahail
+
+!@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
+!
+! Beginning of executable code...
+!
+!@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
+
+  N0r=n0rain
+  N0s=n0snow
+  N0h=n0hail
+  N0g=n0grpl
+
+  rhos=rhosnow
+  rhoh=rhohail
+  rhog=rhogrpl
+
+  alphar = alpharain
+  alphas = alphasnow
+  alphag = alphagrpl
+  alphah = alphahail
+
+  END SUBROUTINE model_dsd
+
+  SUBROUTINE coeff_hail(fw,qml)
```

Comment:
The variable, qml, has been identified as unused.  Please remove unused variable.

Reply:
We changed the subroutine to remove qml.

https://github.com/NOAA-EMC/GSI/pull/142/files#diff-fa0ecb3836c32fcbab1f64a2be08a2292b9c3414d702d475609b32f8205ffcf1R589